tests/scripts/basic-build-test.sh

```diff
@@ -25,13 +25,41 @@
 #
 # This script has been written to be generic and should work on any shell.
 #
-# Usage: basic-build-tests.sh
+# Usage: basic-build-tests.sh               to run all tests
+#        basic-build-tests.sh unit_test     to run only unit tests
 #
 
 # Abort on errors (and uninitiliased variables)
 set -eu
 
-if [ -d library -a -d include -a -d tests ]; then :; else
+CONFIG_H='include/mbedtls/config.h'
+CONFIG_BAK="$CONFIG_H.bak"
+
+IS_UNIT_TEST=0
+
+echo
+if [ -z ${1+x} ]
+then
+    echo "Running all tests"
+else
+    if [ "$1" = "unit_test" ]
+    then
+        echo "Running only unit tests"
+        IS_UNIT_TEST=1
+    fi
+fi
+echo
+
+# Check test environment. If importer, src and inc folders are present then
+# there is a different directory structure and we need to adapt to it.
+if [ -d library -a -d include -a -d tests ] || [ -d importer -a -d inc -a -d src ]
+then
+    if [ -d importer -a -d inc -a -d src ]
```

Comment:
I find the logic a bit unclear with the nested if repeated part of the condition of the top-level if. I think the following structure might be clearer:
```
if [ -d importer -a -d inc -a -d src ]; then
    # adapt config
elif [ -d library -a -d include -a -d tests ]; then :; else
    # complain
fi

Reply:
True, changed as suggested